scale/recipe/models.py

```diff
@@ -1843,43 +1272,18 @@ def edit_recipe_type_v6(self, recipe_type_id, title, description, definition, au
                 msgs = [create_sub_update_recipe_definition_message(id, recipe_type.id) for id in super_ids]
                 CommandMessageManager().send_messages(msgs)
 
-    def get_by_natural_key(self, name, version):
+    def get_by_natural_key(self, name, revision_num):
         """Django method to retrieve a recipe type for the given natural key
 
         :param name: The human-readable name of the recipe type
         :type name: string
+        :param revision_num: The revision number of the recipe type
+        :type revision_num: int
         :returns: The recipe type defined by the natural key
         :rtype: :class:`recipe.models.RecipeType`
         """
 
-        return self.get(name=name, version=version)
-
-    def get_details_v5(self, recipe_type_id):
-        """Gets additional details for the given recipe type model based on related model attributes.
-
-        The additional fields include: job_types.
-
-        :param recipe_type_id: The unique identifier of the recipe type.
-        :type recipe_type_id: int
-        :returns: The recipe type with extra related attributes.
-        :rtype: :class:`recipe.models.RecipeType`
-        """
-
-        # Attempt to fetch the requested recipe type
-        recipe_type = RecipeType.objects.select_related('trigger_rule').get(pk=recipe_type_id)
-
-        # Add associated job type information
-        if RecipeDefinitionSunset.is_seed_dict(recipe_type.definition):
-            # possible to have a v6 definition through a v5 call. Need to get the
-            # job types differently since the v6 RecipeDefinition doesn't have a
-            # get_job_types method
-            job_types = []
-            for job in recipe_type.get_recipe_definition().get_job_type_keys():
-                job_types.append(JobType.objects.get_by_natural_key(name=job.name, version=job.version))
-            recipe_type.job_types = set(job_types)
-        else:
-            recipe_type.job_types = recipe_type.get_recipe_definition().get_job_types()
-        return recipe_type
+        return self.get(name=name, revision_num=revision_num)
```

Comment:
RecipeType should be unique by name alone

Reply:
Removed argument